src/App.js

```diff
@@ -1,25 +1,53 @@
 import React from 'react';
 
 import './App.scss';
+import { Button } from './components/Button/Button';
+import { Clock } from './components/Clock/Clock';
 
-const App = () => {
-  setInterval(() => {
-    const date = new Date();
+class App extends React.Component {
+  state = {
+    isClockVisible: true,
+    currentClockName: 'start name',
+  }
 
+  changeVisibility = () => {
+    this.setState((isClockVisible) => {
+      const prevState = this.state.isClockVisible;
+
+      return { isClockVisible: !prevState };
+    });
+  }
+
+  changeClockName = () => {
+    const oldName = this.state.currentClockName;
+    const newName = Math.random().toString(36).substring(7);
+
+    this.setState({ currentClockName: newName });
     // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
+    console.log(`The Clock was renamed from ${oldName} to ${newName}`);
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+        <div className="clock">
+          {this.state.isClockVisible
+            && <Clock clockName={this.state.currentClockName} />}
+        </div>
+        <Button
+          action={this.changeVisibility}
+          state={this.state.isClockVisible}
```

Comment:
```suggestion
          state={this.state.isClockVisible}
```

state - плохое имя для пропсы, лучше так и назвать "isClockVisible". Человек, который будет смотреть Button компонент никогда не поймет, что state - boolean значение